### PR TITLE
Add more details in the explanation of dist-train.md

### PR DESCRIPTION
**Pull request by chrishkchris**
Add more details in the explanation of dist-train.md

### CODE REVIEW

File: docs-site/docs/dist-train.md [hunk 1]
```diff
@@ -145,30 +165,80 @@ if __name__ == '__main__':
...
 
+Here are some explanations concerning the variables created above:
+
+(i) `nccl_id`
+
+Note that we need to generate a NCCL ID here to be used for collective communication, and then pass it to all the processes. 
+The NCCL ID is like a ticket, where only the processes with this ID can join the AllReduce operation. 
+(Later if we use MPI, the passing of NCCL ID is not necessary, because the ID is broadcased by MPI in our code automatically)
+
+(ii) `num_gpus`
```
Comment (nudles) [inline docs-site/docs/dist-train.md:189]: shall we simplify the terminology by using local_rank and global_rank?

File: docs-site/docs/dist-train.md [hunk 2]
```diff
@@ -75,12 +75,30 @@ sgd = opt.DistOpt(sgd)
...
 
-`dev` represents the `Device` instance, where to load data and run the CNN
+Here are some explanations concerning some variables in the code:
+
+(i) `dev` 
+
+dev represents the `Device` instance, where to load data and run the CNN
 model.
 
+(ii)`rank_in_local`
```
Comment (nudles) [inline docs-site/docs/dist-train.md:85]: local_rank
global_rank

File: docs-site/docs/dist-train.md [hunk 3]
```diff
@@ -20,9 +20,9 @@ the gradients of the weights, which are averaged via All-Reduce (provided by
 [NCCL](https://developer.nvidia.com/nccl)) for weight update following
 stochastic gradient descent algorithms (SGD).
 
-The all-reduce operation by NCCL can be used to reduce and synchronize the
+The All-reduce operation by NCCL can be used to reduce and synchronize the
```
Comment (nudles) [inline docs-site/docs/dist-train.md:23]: All-reduce, All-Reduce, all-reduce?
just be consistent.

File: docs-site/docs/dist-train.md (hunk 1 above)
Comment (chrishkchris) [reply docs-site/docs/dist-train.md:189]: yes, can use <local_rank, global_rank, world size> in every locations of python and C++?

File: docs-site/docs/dist-train.md (hunk 2 above)
Comment (chrishkchris) [reply docs-site/docs/dist-train.md:85]: yes, we use <local_rank, global_rank, world size> everywhere

File: docs-site/docs/dist-train.md (hunk 3 above)
Comment (chrishkchris) [reply docs-site/docs/dist-train.md:23]: ok, I will change

**Comment by chrishkchris** [comment]
I unified the terms in the doc, will change the corresponding python and C communicator code

**Comment by chrishkchris** [comment]
I have already unified terms to <local_rank, global_rank, world_size>

**Pull request merged**